optaplanner-core/src/main/java/org/optaplanner/core/api/solver/ProblemFactChange.java

```diff
@@ -30,8 +31,10 @@
  * So any change must be done on the problem facts and planning entities referenced by the {@link PlanningSolution}
  * of the {@link ScoreDirector}. On each change it should also notify the {@link ScoreDirector} accordingly.
  *
+ * @deprecated in favor of {@link ProblemChange}.
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
+@Deprecated(forRemoval = true)
```

Comment:
We typically also add a line to start the javadoc with "Deprecated in favor of {@link ProblemChange}." I know it's repetition (3 times: first javadoc line + javadoc annotation + java annotation), but it helps to make sure the user doesn't see it :)

Reply:
Ok, but be aware we don't do it consistently across the codebase.